databricks_cli/configure/config.py

```diff
@@ -49,49 +50,73 @@ def decorator(*args, **kwargs):
 
 def get_dbfs_client():
     conf = DatabricksConfig.fetch_from_fs()
-    return DbfsService(ApiClient(conf.username, conf.password, host=conf.host))
+    if conf.is_valid_with_token:
+        return DbfsService(ApiClient(host=conf.host, token=conf.token))
+    return DbfsService(ApiClient(user=conf.username, password=conf.password, host=conf.host))
 
 
 class DatabricksConfig(object):
     home = expanduser('~')
 
-    def __init__(self, host=None, username=None, password=None):
-        self.config = ConfigParser.RawConfigParser()
-        if host:
-            self.config.set(DEFAULT_SECTION, HOST, host)
-        if username:
-            self.config.set(DEFAULT_SECTION, USERNAME, username)
-        if password:
-            self.config.set(DEFAULT_SECTION, PASSWORD, password)
+    def __init__(self):
+        self._config = ConfigParser.RawConfigParser()
 
     def overwrite(self):
-        config_path = join(self.home, '.databrickscfg')
+        config_path = self.get_path()
         with open(config_path, 'wb') as cfg:
-            self.config.write(cfg)
+            self._config.write(cfg)
         os.chmod(config_path, 0o600)
 
     @property
     def is_valid(self):
-        return self.config.has_option(DEFAULT_SECTION, USERNAME) and \
-            self.config.has_option(DEFAULT_SECTION, PASSWORD) and \
-            self.config.has_option(DEFAULT_SECTION, HOST)
+        return self.is_valid_with_password or self.is_valid_with_token
+
+    @property
+    def is_valid_with_password(self):
+        return self._config.has_option(DEFAULT_SECTION, USERNAME) and \
+            self._config.has_option(DEFAULT_SECTION, PASSWORD) and \
+            self._config.has_option(DEFAULT_SECTION, HOST)
+
+    @property
+    def is_valid_with_token(self):
+        return self._config.has_option(DEFAULT_SECTION, TOKEN) and \
+            self._config.has_option(DEFAULT_SECTION, HOST)
 
     @property
     def host(self):
-        return self.config.get(DEFAULT_SECTION, HOST) if self.is_valid else None
+        return self._config.get(DEFAULT_SECTION, HOST) if self.is_valid else None
```

Comment:
I don't see why we need is_valid check here.

Reply:
The checks seem to be necessary here. If the key does not exist the get function will raise a `NoOptionError` exception.